### DISCUSSION TITLE
AI Deck not flashing GAP8

### DISCUSSION THREAD

**Discussion by INSACrazyflie** [bitcraze/discussions#1048] [Q&A] [answered]
Hello,

For a university project, we recently purchased an AI deck to use for
vision algorithms on the Crazyflie2.1.
Unfortunately we didn't manage to flash code on the deck via wifi (this was one of the features we expected from the deck). Do you have a solution?

The deck is brand new and seems functional apart from the above problem.

 Yours faithfully

**Comment by tobbeanton** [comment, marked as answer]
Unfortunately you are going to need a [debugger](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/infrastructure/jtag-programmer/), which is also needed to do efficient development on it.

**Reply by INSACrazyflie** [reply]
So I'll have to buy a debugger for a single use? Or does the debugger greatly simplify the deployment of code on the deck? Because being able to flash code via wifi was one of the motivations for buying the AI deck. I don't really want to buy a debugger for a single use, since the deck is already expensive.

**Reply by ClementSrs** [reply]
You have to connect your crazyflie's AI-Deck (GAP8 port) with OLimex-20-10 and a Jlink then follow the instructions of the tutorial.

**Reply by tobbeanton** [reply]
Yes a debugger is necessary as written in the product page.

**Reply by INSACrazyflie** [reply]
Yes, my bad I miss read sorry 😅
